wingtips-core/src/main/java/com/nike/wingtips/TraceAndSpanIdGenerator.java

```diff
@@ -34,15 +38,47 @@ private TraceAndSpanIdGenerator() {
     }
 
     /**
-     * @return A newly-generated random 64-bit long as a String.
+     * @return A newly-generated random 64-bit long encoded as a String <b>UNSIGNED AND IN HEX FORMAT</b> - intended for use as a trace or span ID.
+     *          The returned string will have a length of 16 characters (zeroes will be prepended as padding if necessary to
+     *          reach a string length of 16). The hex format is necessary to be fully B3 compatible and therefore fully Zipkin compatible
+     *          (see <a href="http://zipkin.io/pages/instrumenting.html">http://zipkin.io/pages/instrumenting.html</a>). If you simply want a full 64-bit
+     *          random normal *signed* Java long encoded in *decimal* format for other reasons then you can call {@link String#valueOf(long)} and pass in a
+     *          long created by {@link #generate64BitRandomLong()}.
+     *          <p>You can convert the unsigned hex-encoded longs returned by this method back into normal Java long primitives by passing them to
+     *          {@link #unsignedLowerHexStringToLong(String)}.
      */
     public static String generateId() {
+        return longToUnsignedLowerHexString(generate64BitRandomLong());
+    }
+
+    /**
+     * @return A random long pulled from the full 64-bit random search space (as opposed to the 48 bits of randomness you get from
+     *          {@link java.util.Random#nextLong()}).
+     */
+    public static long generate64BitRandomLong() {
         byte[] random8Bytes = new byte[8];
```

Comment:
I'll keep it in mind for future optimization investigations. For now I've never noticed a problem (and it's what UUID does under the hood, so I assume if there was a noticeable benefit to that optimization it would have been done there already - possibly incorrect assumption, but still).